aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/Orchestrator.java

```diff
@@ -95,22 +98,38 @@ public Orchestrator(
      * @return A Completable operation to start the sync engine orchestrator
      */
     @NonNull
-    public Completable start() {
-        return Completable.fromAction(() -> {
-            storageObserver.startObservingStorageChanges();
-            subscriptionProcessor.startSubscriptions();
-            syncProcessor.hydrate().blockingAwait();
-            mutationProcessor.startDrainingMutationOutbox();
-            subscriptionProcessor.startDrainingMutationBuffer();
+    public synchronized Completable start() {
+        return Completable.fromSingle(single -> {
+            try {
+                if (!storageObserver.isObserving()) {
```

Comment:
Let's align the naming of all of these to their corresponding `start<whatever>` methods. Here, this would be come:
```
storageObserver.isObservingStorageChanges()
```